cmd/filestatus.go

```diff
@@ -0,0 +1,255 @@
+package cmd
+
+import (
+	"crypto/md5" //nolint:gosec
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/dustin/go-humanize" //nolint:misspell
+	"github.com/spf13/cobra"
+	"github.com/wtsi-hgi/ibackup/put"
+	"github.com/wtsi-hgi/ibackup/set"
+	"github.com/wtsi-npg/extendo/v2"
+)
+
+// options for this cmd.
+var filestatusIrods bool
+var filestatusDB string
+
+// statusCmd represents the status command.
+var filestatusCmd = &cobra.Command{
+	Use:   "filestatus",
+	Short: "Get the status of a file in the database",
+	Long: `Get the status of a file in the database.
+
+Prints out a summary of the given file for each set the file appears in.
+
+The --database option should be the path to the local backup of the iBackup
+database, defaulting to the value of the IBACKUP_DATABASE_PATH environmental
+variable.
+
+The --irods options will gather additional information about the file, such as
+the local and remote checksums.
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) < 1 {
+			die("you must supply the file to be checked")
+		}
+
+		err := fileSummary(filestatusDB, args[0], filestatusIrods)
+		if err != nil {
+			die(err.Error())
+		}
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(filestatusCmd)
+
+	filestatusCmd.Flags().StringVarP(&filestatusDB, "database", "d",
+		os.Getenv("IBACKUP_DATABASE_PATH"), "path to iBackup database file")
+	filestatusCmd.Flags().BoolVarP(&filestatusIrods, "irods", "i", false,
+		"do additional checking in iRods")
+}
+
+func fileSummary(dbPath, filePath string, useIrods bool) error {
+	db, err := set.NewRO(dbPath)
+	if err != nil {
+		return err
+	}
+
+	fsg := newFSG(db, filePath, useIrods)
```

Comment:
Move this below the db.GetAll so it's next to where it's first used, and we don't open a connection to irods if the db call fails.